src/examples/learn_quant/quantifier.py

```diff
@@ -0,0 +1,102 @@
+
+@dataclass(eq=True, frozen=True)
+class QuantifierModel(Referent):
+    """A quantifier model is a single referent that captures a particular interpretation of a quantifier meaning, which is a set of quantifier referents.
+
+    Every quantifier model is a quadruple <M, A, B>, where M corresponds to all possible quantifier referents for a given communicative situation, A and B are differents sets of quantifier referents that correspond to the items of comparison in quantificational logic.
+    
+    X denotes the set of all possible quantifier referents in a given Universe. A and B are subsets of M and each of M, A, and B are subsets of X. 
+
+    For the purposes of this project, a QuantifierModel is a Referent (capital R), but the individual referents for a given model are the indices of the QuantifierModel's sets.
+
+    0 => A
+    1 => B
+    2 => A | B
+    3 => M - (A | B)
+    4 => X - (M | A | B)
+    
+    """
+
+    name: str = None
+    M: frozenset = field(init=False)
+    A: frozenset = field(init=False)
+    B: frozenset = field(init=False)
+
+    def __post_init__(self):
+        object.__setattr__(self, 'A', frozenset([i for i, x in enumerate(self.name) if x in ['0','2']]))
+        object.__setattr__(self, 'B', frozenset([i for i, x in enumerate(self.name) if x in ['1','2']]))
+        object.__setattr__(self, 'M', frozenset([i for i, x in enumerate(self.name) if x in ['0','1','2','3']]))
+
+    @classmethod
+    def from_sets(cls, M: set | frozenset, A: set | frozenset, B: set | frozenset):
+        return cls(name=None, M=frozenset(M), A=frozenset(A), B=frozenset(B))
+
+    def get_cardinalities(self) -> dict:
+        return {"M": len(self.M), 
+                "A": len(self.A), 
+                "B": len(self.B), 
+                }
+    
+    def to_numpy(self, quantifier_index=None, in_meaning=False):
+        # Convert the string to an array of integers
+        indices = np.fromiter(self.name, dtype=int)
+        
+        # Initialize a zero matrix with shape (len(s), 5)
+        one_hot_array = np.zeros((len(indices), 5), dtype=int)
+        
+        # Use numpy advanced indexing to set the appropriate elements to 1
+        one_hot_array[np.arange(len(indices)), indices] = 1
+
+        # If quantifier_index is provided, concatenate it to each vector in one_hot_array
+        if quantifier_index is not None:
+            # Ensure quantifier_index is an array
+            quantifier_index = np.asarray(quantifier_index)
+            if quantifier_index.ndim == 1:
+                # Concatenate quantifier_index to each vector
+                quantifier_index = quantifier_index.reshape(1, -1).repeat(len(indices), axis=0)
+                one_hot_array = np.hstack((one_hot_array, quantifier_index))
+            else:
+                raise ValueError("quantifier_index must be a one-dimensional one-hot vector.")
+        else:
+            appended_value = 0
+            if in_meaning:
```

Comment:
Not convinced this `in_meaning` stuff should be part of this method, versus handled elswhere in the data processing, i.e. wherever this is being called from.  Because when supplying to the model, the label 1/0 is going to be separate from this array anywyas

Reply:
To be more explicit: the combination of model + quantifier label are inputs (i.e. `x`) to a model, and the truth value (which I think is what `in_meaning` is doing) is the goal/output/`y` value